block-producer/src/server/mod.rs

```diff
@@ -52,19 +52,15 @@ pub async fn serve(config: BlockProducerConfig) -> Result<()> {
 
     let block_producer = api_server::ApiServer::new(api::BlockProducerApi::new(queue.clone()));
 
-    tokio::spawn(async move {
-        queue
-            .run()
-            .instrument(info_span!(target: COMPONENT, "transaction_queue_start"))
-            .await
-    });
+    tokio::spawn(
+        async move { queue.run().await }
+            .instrument(info_span!(target: COMPONENT, "transaction_queue")),
+    );
```

Comment:
I'm not sure we need to instrument this. I think batch building methods are named explicitly enough so that we won't confuse them with something else.